model/labels/labels_internlabels.go

```diff
@@ -0,0 +1,778 @@
+		for j < len(names) && names[j] < name {
+			j++
+		}
+		if name == MetricName || (j < len(names) && name == names[j]) {
+			continue
+		}
+		b = append(b, name...)
+		b = append(b, seps[0])
+		b = append(b, value...)
+		b = append(b, seps[0])
+	}
+	return xxhash.Sum64(b), b
+}
+
+// BytesWithLabels is just as Bytes(), but only for labels matching names.
+// 'names' have to be sorted in ascending order.
+func (ls Labels) BytesWithLabels(buf []byte, names ...string) []byte {
+	b := bytes.NewBuffer(buf[:0])
+	j := 0
+	for pos := 0; pos < len(ls.data); {
+		lName, newPos := decodeString(ls.syms, ls.data, pos)
+		lValue, newPos := decodeString(ls.syms, ls.data, newPos)
+		for j < len(names) && names[j] < lName {
+			j++
+		}
+		if j == len(names) {
+			break
+		}
+		if lName == names[j] {
+			if b.Len() > 1 {
+				b.WriteByte(seps[0])
+			}
+			b.WriteString(lName)
+			b.WriteByte(seps[0])
+			b.WriteString(lValue)
+		}
+		pos = newPos
+	}
+	return b.Bytes()
+}
+
+// BytesWithoutLabels is just as Bytes(), but only for labels not matching names.
+// 'names' have to be sorted in ascending order.
+func (ls Labels) BytesWithoutLabels(buf []byte, names ...string) []byte {
+	b := bytes.NewBuffer(buf[:0])
+	j := 0
+	for pos := 0; pos < len(ls.data); {
+		lName, newPos := decodeString(ls.syms, ls.data, pos)
+		lValue, newPos := decodeString(ls.syms, ls.data, newPos)
+		for j < len(names) && names[j] < lName {
+			j++
+		}
+		if j == len(names) || lName != names[j] {
+			if b.Len() > 1 {
+				b.WriteByte(seps[0])
+			}
+			b.WriteString(lName)
+			b.WriteByte(seps[0])
+			b.WriteString(lValue)
+		}
+		pos = newPos
+	}
+	return b.Bytes()
+}
+
+// Copy returns a copy of the labels.
+func (ls Labels) Copy() Labels {
+	buf := append([]byte{}, ls.data...)
+	return Labels{syms: ls.syms, data: yoloString(buf)}
+}
+
+// Get returns the value for the label with the given name.
+// Returns an empty string if the label doesn't exist.
+func (ls Labels) Get(name string) string {
+	if name == "" { // Avoid crash in loop if someone asks for "".
+		return "" // Prometheus does not store blank label names.
+	}
+	for i := 0; i < len(ls.data); {
+		var lName, lValue string
+		lName, i = decodeString(ls.syms, ls.data, i)
+		if lName == name {
+			lValue, _ = decodeString(ls.syms, ls.data, i)
+			return lValue
+		} else if lName[0] > name[0] { // Stop looking if we've gone past.
+			break
+		}
+		_, i = decodeVarint(ls.data, i)
+	}
+	return ""
+}
+
+// Has returns true if the label with the given name is present.
+func (ls Labels) Has(name string) bool {
+	if name == "" { // Avoid crash in loop if someone asks for "".
+		return false // Prometheus does not store blank label names.
+	}
+	for i := 0; i < len(ls.data); {
+		var lName string
+		lName, i = decodeString(ls.syms, ls.data, i)
+		if lName == name {
+			return true
+		} else if lName[0] > name[0] { // Stop looking if we've gone past.
+			break
+		}
+		_, i = decodeVarint(ls.data, i)
+	}
+	return false
```

Comment:
Sounds like this entire method could be just this, with no performance impact:
```suggestion
	return ls.Get(name) != ""
```

Reply:
Interesting and surprising. The impact seems to be a quite consistent 4ns increase, right? I guess that's the cost of the extra `decodeString` call for getting the value.